test/integration/header_integration_test.cc

```diff
@@ -90,6 +99,57 @@ class HeaderIntegrationTest : public HttpIntegrationTest,
     header_value_option->mutable_append()->set_value(append);
   }
 
+  void prepareEDS() {
+    config_helper_.addConfigModifier([&](envoy::api::v2::Bootstrap& bootstrap) {
+      auto* static_resources = bootstrap.mutable_static_resources();
+      ASSERT(static_resources->clusters_size() == 1);
+
+      static_resources->mutable_clusters(0)->CopyFrom(
+          TestUtility::parseYaml<envoy::api::v2::Cluster>(
+              R"EOF(
+                  name: cluster_0
+                  type: EDS
+                  eds_cluster_config:
+                    eds_config:
+                      api_config_source:
+                        cluster_name: "eds-cluster"
+                        api_type: GRPC
+              )EOF"));
+
+      // Give the config utility a place to stuff the upstream host's port (must come before
```

Comment:
This is a little confusing. Is this just because of how the default config stuff works? Would it be worth it to build this EDS stuff into the base integration class? It seems generally useful and very cool. cc @alyssawilk

Reply:
Yeah, I think the same way that we special cased the HttpConnectionManager it'd be useful to have utilities to play with EDS config directly.

Reply:
I can take a look at making the config smarter about EDS in another review, if you'd like.

Reply:
If you want to do a follow up / TODO that's fine, but can you add more of a comment here? It's confusing why the placeholder is necessary.

Reply:
But yes, it's to do with ConfigHelper expecting the number of ports assigned to match the number of  upstreams during the call to finalize.